app/services/bus_notice_service.py

```
@@ -0,0 +1,346 @@
+from typing import List, Dict, Optional, Any
+
+from app.config.settings import settings
+from app.services.bus_logic.restricted_bus import TOPISCrawler
+from app.services.bus_logic.position_checker import get_stations_by_position
+
+logger = logging.getLogger(__name__)
+KST = pytz.timezone('Asia/Seoul')
+
+class BusNoticeService:
+    crawler: Optional[TOPISCrawler] = None
+    cached_notices: List[Dict] = []
```

Ruff (0.15.1)

[warning] Mutable default value for class attribute

(RUF012)